src/app/filter/filter.component.html

```diff
@@ -0,0 +1,39 @@
+<div class="main-box">
+  <div *ngIf="shouldShow" class="filter-box">
+      <div [matMenuTriggerFor]="filterMenu" class="filter-text-icon">
+        <mat-icon class="filter-icon">filter_list</mat-icon>Filter
+      </div>
+      <button mat-icon-button class="filter-icon filter-reset" (click)="resetFilter()">
+        <mat-icon *ngIf="filterService.isFiltering()">close</mat-icon>
+      </button>
+  </div>
+
+  <mat-menu #filterMenu="matMenu">
+    <div (click)="$event.stopPropagation();" class="menu-box">
+      <div class="title-filter-box">
+        <mat-form-field appearance="outline" class="filter-title-input">
+          <mat-label>Title</mat-label>
+          <mat-hint>Task Title</mat-hint>
+          <input matInput (input)="titleFilter($event.target.value)" placeholder="Title" autocomplete="off">
+        </mat-form-field>
+      </div>
+      <div class="select-label-box">
+        <mat-checkbox (change)="noLabelFilter($event.checked)"></mat-checkbox>
+        <div class="label-name">No label</div>
+      </div>
+      <div *ngFor="let label of labels" class="select-label-box">
+        <mat-checkbox (change)="labelFilter($event.checked, label)"></mat-checkbox>
+        <div class="label" [style]="'background-color:' + label.color"></div>
```

Comment:
```suggestion
        <div class="label" [style.background-color]="label.color"></div>
```
Same for other files